### PR TITLE
stats: speed up loading stats info when server starts

### LINKED ISSUE #5194: Speed up loading statistic info
If there are many tables in TiDB, loading stats could be quite slow. We should make it fast.
See: https://github.com/pingcap/tidb/pull/5188#issuecomment-346414770

### PR DESCRIPTION
**Pull request by lamxTyler**
The `Update` updates the stats cache using delta load strategy, it might be quite slow when server just starts.
Fix #5194

### CODE REVIEW

**Comment by hanfei1991** [comment]
We can new a file named `bootstrap.go`

**Comment by shenli** [comment]
Could you provide some benchmark result?

**Comment by lamxTyler** [comment]
@shenli On my local machine, with 800~ tables, each with 10 columns, 3 indices, it takes about 1.5s to load with this pr, while master takes about 10.8s.

File: domain/domain.go [hunk 1]
```diff
@@ -609,6 +609,10 @@ func (do *Domain) updateStatsWorker(ctx context.Context, lease time.Duration) {
 	loadHistogramTicker := time.NewTicker(lease)
 	defer loadHistogramTicker.Stop()
 	statsHandle := do.StatsHandle()
+	err := statsHandle.InitStats(do.InfoSchema())
+	if err != nil {
+		log.Error("[stats] init stats info fail: ", errors.ErrorStack(err))
```
Comment (zz-jason) [inline domain/domain.go]: s/fail/failed/

File: statistics/boostrap.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/terror"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/sqlexec"
+)
+
+func (h *Handle) initStatsMeta(is infoschema.InfoSchema) (statsCache, error) {
+	sql := fmt.Sprintf("select version, table_id, modify_count, count from mysql.stats_meta")
```
Comment (zz-jason) [inline statistics/boostrap.go]: no need to `Sprintf"

File: statistics/boostrap.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+			ModifyCount: row.GetInt64(2),
+			Version:     row.GetUint64(0),
+		}
+		tables[tableID] = tbl
+	}
+	return tables, nil
+}
+
+func (h *Handle) initStatsHistograms(is infoschema.InfoSchema, tables statsCache) error {
+	sql := fmt.Sprintf("select table_id, is_index, hist_id, distinct_count, version, null_count, cm_sketch from mysql.stats_histograms")
```
Comment (zz-jason) [inline statistics/boostrap.go]: ditto

File: statistics/boostrap.go [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+				continue
+			}
+			table.Columns[hist.ID] = &Column{Histogram: hist, Info: colInfo}
+		}
+	}
+	return nil
+}
+
+func (h *Handle) initStatsBuckets(tables statsCache) error {
+	sql := fmt.Sprintf("select table_id, is_index, hist_id, bucket_id, count, repeats, lower_bound, upper_bound from mysql.stats_buckets")
```
Comment (zz-jason) [inline statistics/boostrap.go]: ditto

File: statistics/boostrap.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+			}
+			column.Count += row.GetInt64(4)
+			if !mysql.HasPriKeyFlag(column.Info.Flag) {
+				continue
+			}
+			hist = &column.Histogram
+			d := row.GetDatum(6, &fields[6].Column.FieldType)
+			lower, err = d.ConvertTo(h.ctx.GetSessionVars().StmtCtx, &column.Info.FieldType)
+			if err != nil {
+				terror.Log(errors.Trace(err))
```
Comment (XuHuaiyu) [inline statistics/boostrap.go]: should we record 
a more detailed msg here.
same as line 148

File: statistics/boostrap.go [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+		for _, col := range table.Columns {
+			for i := 1; i < len(col.Buckets); i++ {
+				col.Buckets[i].Count += col.Buckets[i-1].Count
+			}
+		}
+	}
+	return nil
+}
+
+// InitStats will init the stats cache using a faster strategy than `Update` which is used for delta update.
```
Comment (XuHuaiyu) [inline statistics/boostrap.go]: a faster strategy is not explict,
 use `full load` directly?

**Pull request merged**